server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -724,4 +756,86 @@ public int getConsumerEntitlementCount(Owner owner, ConsumerType type) {
             .setResultTransformer(Criteria.DISTINCT_ROOT_ENTITY)
             .list();
     }
+    /**
+     * InExpressionIgnoringCase
+     *
+     * Allows Criterion for case insensitive comparison to list of values
+     *
+     * @author wpoteat
+     */
+    public class InExpressionIgnoringCase implements Criterion {
+
+        private final String propertyName;
+        private final Object[] values;
+
+        public InExpressionIgnoringCase(final String propertyName, final Object[] values) {
+            this.propertyName = propertyName;
+            this.values = values;
+        }
+
+        public String toSqlString(final Criteria criteria, final CriteriaQuery criteriaQuery)
```

Comment:
When building large strings like this, use a proper StringBuilder. Java 1.7+ will somewhat do this for you, but it does so by creating several instances unnecessarily, when you could do it quicker/more efficiently by handling it yourself.